src/main/java/org/fogbowcloud/manager/core/threads/OpenProcessor.java

```diff
@@ -0,0 +1,159 @@
+package org.fogbowcloud.manager.core.threads;
+
+import java.util.Properties;
+
+import org.apache.log4j.Logger;
+import org.fogbowcloud.manager.core.constants.ConfigurationConstants;
+import org.fogbowcloud.manager.core.constants.DefaultConfigurationConstants;
+import org.fogbowcloud.manager.core.instanceprovider.InstanceProvider;
+import org.fogbowcloud.manager.core.models.linkedList.ChainedList;
+import org.fogbowcloud.manager.core.models.orders.Order;
+import org.fogbowcloud.manager.core.models.orders.OrderState;
+import org.fogbowcloud.manager.core.models.orders.instances.OrderInstance;
+
+public class OpenProcessor implements Runnable {
+
+	private InstanceProvider localInstanceProvider;
+	private InstanceProvider remoteInstanceProvider;
+
+	private ChainedList openOrdersList;
+	private ChainedList pendingOrdersList;
+	private ChainedList failedOrdersList;
+	private ChainedList spawningOrdersList;
+
+	private String localMemberId;
+
+	/**
+	 * Attribute that represents the thread sleep time when there is no Orders
+	 * to be processed.
+	 */
+	private Long sleepTime;
+
+	private static final Logger LOGGER = Logger.getLogger(OpenProcessor.class);
+
+	public OpenProcessor(InstanceProvider localInstanceProvider, InstanceProvider remoteInstanceProvider,
+			String localMemberId, Properties properties) {
+		this.localInstanceProvider = localInstanceProvider;
+		this.remoteInstanceProvider = remoteInstanceProvider;
+		this.localMemberId = localMemberId;
+
+		// TODO: ChainedLists instanciation by Singleton pattern.
+
+		String schedulerPeriodStr = properties.getProperty(ConfigurationConstants.OPEN_ORDERS_SLEEP_TIME_KEY,
+				DefaultConfigurationConstants.OPEN_ORDERS_SLEEP_TIME);
+		this.sleepTime = Long.valueOf(schedulerPeriodStr);
+	}
+
+	@Override
+	public void run() {
+		while (true) {
+			try {
+				Order order = this.openOrdersList.getNext();
+				if (order != null) {
+					this.processOpenOrder(order);
+				} else {
+					LOGGER.info("There is no Open Order to be processed, sleeping OpenProcessor Thread...");
```

Comment:
there is no need to add the class name in the log. So, "There is no open order to be processed, sleeping for " + this.sleepTime

Reply:
Done, [here](https://github.com/fogbow/fogbow-manager-core/pull/13/commits/2c530ad051c2f728874872ffbdc26a8b19dc9bd1).